napari/layers/labels/labels.py

```diff
@@ -710,7 +710,7 @@ def color_mode(self, color_mode: Union[str, LabelColorMode]):
             self._colormap = ensure_colormap(self._direct_colormap)
         self._selected_color = self.get_color(self.selected_label)
         self.events.color_mode()
-        self.events.colormap()
+        self.events.colormap()  # If remove this emitting, connect shader update to color_mode
```

Comment:
Why this choice? It seems better to have more informative events, and just handle them the same downstream, instead of emitting the same event here.

Reply:
Because the calculation hash map is not trivial (and not so fast), and I have not found another way to do this single. 

Also, real change is not connecting to the `color_mode` event in VispyLayer. And this comment is to reconnect in such a situation.

Reply:
Yes I saw, that's what I mean. The change basically doesn't change how many times the shader is updated, because anyways you're connecting to `layer.events.colormap()` right?